misc/test-client-rust/src/util.rs

```diff
@@ -52,16 +49,15 @@ pub fn make_request(
     }
 
     if status == StatusCode::NO_CONTENT {
-        return (HashMap::new(), status);
+        return (serde_json::json!({}), status);
     }
 
-    let response_fields = response
-        .json::<Response>()
-        .expect("Failed to parse response as HashMap<String, Value>");
+    let response_text = response.text().expect("Failed to get response text");
 
-    verbose_log(&args, format!("response: {:?}", response_fields).as_str());
+    let response_value = serde_json::from_str::<Value>(&response_text).expect("Failed to parse JSON response");
```

Comment:
Instead of calling `response.text()` and then `serde_json::from_str`, we can just call `response.json::<Value>()`